src/interop.md

```diff
@@ -32,30 +32,158 @@ We'll call this API wrapping - the API wrapper takes care of:
 * introducing Nim idioms such as generics
 * adapting the [error handling](./errors.md) model
 
-### Memory
+### Calling Nim code from other languages
+
+When calling Nim from other languages, care must be taken to first initialize the garbage collector, at least once for every thread - this can be done in two ways:
+
+* Exporting a dedicated "initialization function" for the library:
+  ```nim
+  # Use a dedicate initialization function - the application will crash if users of the library forget to call this function!
+  proc initializeMyLibrary() {.exportc.} =
+    setupForeignThreadGc()
```

Comment:
This is not a good advice. 

When you init the Nim GC, it stores the current stack height as a "bottom of the stack". Later, this value will be used as a starting point for the conservative stack scanning (Nim will iterate from the stored "bottom" value to the current "top" value). Thus, it's safe to call `setupForeignThreadGc` only when you can ensure that it was called from the minimum possible stack height in your application.

Reply:
hm, this makes the recommendation a lot more difficult - it practically mandates "explicit init", which in turn means we need to add an "nim is uninitialized" error to every exported function :/

so more precisely, the issue would be that:

* foreign code calls nim from some deep stack function
* foreign code calls nim from shallow function, where nim allocates gc'd object
* the object allocated in the shallof function is not stack-scanned in a subsequent allocation

?

Reply:
Fortunately for us, `nimGC_setStackBottom` already contains the directional protection, so all that's needed is the helper that declares a variable and uses its address.

I'm starting to think that perhaps the best way out of this mess is `stew/ffi` which contains something like `{.exportffi.}` that replaces `{.exportc.}` and does all this stuff..